Automattic/vip-cli
# Add Feature to Support HTTPS/HTTP/NO_PROXY Settings from System

## Description

**Goal:** To allow individuals operating within non-SOCKS corporate proxies (namely HTTP/HTTPS) to use the VIP-CLI since node-fetch does not "respect" system environment variables

- Added "feature" environment variable VIP_PROXY_OTHER_ENABLED to activate this functionality (while keeping the library backwards compatible for those who don't use a proxy)
- Once enabled, all requests that originally used the SocksProxyAgent will use a proxy agent based on their respective environment variables (including NO_PROXY)
- Allows for upper or lowercase environment variables to be specified (upper case takes precedence)
- The VIP_PROXY functionality is unchanged. Regardless of the existence of any other environment variable, VIP_PROXY takes precedence no matter what

## Steps to Test

Outline the steps to test and verify the PR here.

1. Check out PR.
2. Run `npm install`
3. Run `npm run build`
4. Run `./dist/bin/vip.js` with your favorite commands (too many here to mention)
5. Verify commands run as expected given the environment variables set on your system

**Note: I realize this is an "impactful" change given how often web requests are made. That being said, I am not sure the best way to test this with confidence since everyone works in a slightly different environment proxy wise. So any thoughts on that would be appreciated.**

## Potential Related Work

https://github.com/Automattic/vip/tree/add/proxy-support
https://github.com/Automattic/vip/tree/renovate/socks-proxy-agent-6.x
